src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -151,41 +143,151 @@ private Task ConsumeChunkBytesAsync(ulong bytesConsumed)
                 }
             }
 
-            public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+            private async Task ConsumeTrailingHeaders()
```

Comment:
I wonder if you can move this logic to ReadChunkFromConnectionBuffer, so that you don't need to handle the state specifically in ReadAsync/CopyToAsync. We're already switching on the state there.  Seems like we could just add another case for this state and consume the trailers.  We'd need a way to indicate EOF from ReadChunkFromConnectionBuffer, but that seems doable -- either use -1 or change to a bool TryReadChunk(..., out bytesRead) pattern.

Reply:
Ok